crates/holochain/src/conductor/interface.rs

```diff
@@ -1,145 +1,36 @@
-use crate::conductor::api::*;
-use error::InterfaceResult;
-use futures::{
-    channel::mpsc::{channel, Receiver, Sender},
-    future::{BoxFuture, FutureExt},
-    sink::{Sink, SinkExt},
-    stream::{Stream, StreamExt},
-};
-use holochain_serialized_bytes::{SerializedBytes, SerializedBytesError};
+use crate::{conductor::api::*, core::signal::Signal};
+use error::{InterfaceError, InterfaceResult};
 use serde::{Deserialize, Serialize};
-use std::convert::{TryFrom, TryInto};
+use std::convert::TryInto;
+use tokio::sync::broadcast;
 
 pub mod error;
 pub mod websocket;
 
-/// Allows the conductor or cell to forward signals to connected clients
-pub struct ConductorSideSignalSender<Sig>
-where
-    Sig: 'static + Send + TryInto<SerializedBytes, Error = SerializedBytesError>,
-{
-    sender: Sender<SerializedBytes>,
-    phantom: std::marker::PhantomData<Sig>,
-}
+#[derive(Clone, Debug)]
+pub struct SignalBroadcaster(Vec<broadcast::Sender<Signal>>);
```

Comment:
I'm not exactly sure why you have are many Senders here. Broadcasts allow the Senders to be cloned. Would that not be better then creating multiple broadcasts? I might be missing something here?

Reply:
@freesig there's one for each interface. Each interface has its own broadcast::Sender, but this is the collection of all senders for all interfaces. I wanted to keep the interfaces separate even though right now they all receive all messages, because in the future we'll have more fine-grained control over signals, and perhaps the choice of interface will actually matter.